models/migrations/v231.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"fmt"
+
+	"code.gitea.io/gitea/models/webhook"
+	"code.gitea.io/gitea/modules/json"
+	"code.gitea.io/gitea/modules/secret"
+	"code.gitea.io/gitea/modules/setting"
+	api "code.gitea.io/gitea/modules/structs"
+
+	"xorm.io/builder"
+	"xorm.io/xorm"
+)
+
+func batchProcess[T any](x *xorm.Engine, query func() *xorm.Session, buf []T, process func(*xorm.Session, T) error) error {
```

Comment:
Move the two _bigger_ functions as last two arguments.
```suggestion
func batchProcess[T any](x *xorm.Engine, buf []T, query func() *xorm.Session, process func(*xorm.Session, T) error) error {
```